CMakeLists.txt

```diff
@@ -352,9 +351,22 @@ if(target MATCHES "x64")
       -maes
       -mpclmul
       -mavx
-      -mavx2
```

Comment:
`-mavx2` should indeed be removed, as all CPUs don't support AVX2, and applying that flag globally would cause AVX2 code to be used even in functions where AVX2 support is not checked at runtime.

Reply:
Not setting that causes errors like the following unfortunately:

```
always_inline function '_mm256_xor_si256' requires target feature 'avx2', but would be inlined into function 'blake2b_compress_avx2' that is compiled without support for 'avx2'
```

Reply:
That's why I initially enabled all the target features for which we declared headers being available.

Reply:
Looks like the bundled libsodium version is... 5 years old? Maybe update it to the current `stable`.